src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.Map.cs

```diff
@@ -60,5 +69,109 @@ public void ReadEndMap()
+            {
+                ValidateKeyUniqueness(currentKeyRange);
+            }
+
+            _curentItemIsKey = false;
+        }
+
+        private void HandleMapValueAdded()
+        {
+            Debug.Assert(_currentKeyOffset != null && !_curentItemIsKey);
+
+            _currentKeyOffset = _bytesRead;
+            _curentItemIsKey = true;
+        }
+
+        private void ValidateSortedKeyEncoding((int offset, int length) currentKeyRange)
+        {
+            Debug.Assert(_currentKeyOffset != null);
+
+            if (_previousKeyRange != null)
+            {
+                (int offset, int length) previousKeyRange = _previousKeyRange.Value;
+
+                ReadOnlySpan<byte> originalBuffer = _originalBuffer.Span;
+                ReadOnlySpan<byte> previousKeyEncoding = originalBuffer.Slice(previousKeyRange.offset, previousKeyRange.length);
+                ReadOnlySpan<byte> currentKeyEncoding = originalBuffer.Slice(currentKeyRange.offset, currentKeyRange.length);
+
+                int cmp = CborConformanceLevelHelpers.CompareEncodings(previousKeyEncoding, currentKeyEncoding, ConformanceLevel);
+                if (cmp > 0)
+                {
+                    throw new FormatException("CBOR map keys are not in sorted encoding order.");
+                }
+                else if (cmp == 0 && CborConformanceLevelHelpers.RequiresUniqueKeys(ConformanceLevel))
+                {
+                    throw new FormatException("CBOR map contains duplicate keys.");
+                }
+            }
+
+            _previousKeyRange = currentKeyRange;
+        }
+
+        private void ValidateKeyUniqueness((int offset, int length) currentKeyRange)
+        {
+            Debug.Assert(_currentKeyOffset != null);
+
+            HashSet<(int offset, int length)> previousKeys = GetPreviousKeyRanges();
+
+            if (!previousKeys.Add(currentKeyRange))
+            {
+                throw new FormatException("CBOR map contains duplicate keys.");
+            }
+        }
+
+        private HashSet<(int offset, int length)> GetPreviousKeyRanges()
+        {
+            if (_previousKeyRanges == null)
+            {
+                _keyComparer ??= new KeyEncodingComparer(this);
+                _previousKeyRanges = new HashSet<(int offset, int length)>(_keyComparer);
+            }
+
+            return _previousKeyRanges;
+        }
+
+        private ReadOnlySpan<byte> GetKeyEncoding((int offset, int length) range)
+        {
+            return _originalBuffer.Span.Slice(range.offset, range.length);
+        }
+
+        private class KeyEncodingComparer : IEqualityComparer<(int offset, int length)>
+        {
+            private readonly CborReader _reader;
+
+            public KeyEncodingComparer(CborReader reader)
+            {
+                _reader = reader;
+            }
+
+            public int GetHashCode((int offset, int length) value)
```

Comment:
Naming: Tuple elements use PascalCase.

```suggestion
            public int GetHashCode((int Offset, int Length) value)
```

Reply:
I kind of always assumed we were using camelCase 🤷

Reply:
We had a very long discussion about https://github.com/dotnet/runtime/blob/98d9e7e10330ab6a6e9c854f98b484e803ebb55d/src/libraries/System.Private.CoreLib/src/System/Range.cs#L111 and the next update of Framework Design Guidelines will say

* DO name elements of a named tuple using PascalCase, as if they were properties.

Reply:
I also find camelCase more natural but if we've set on that then let it be